examples/AspNetCore/Controllers/WeatherForecastController.cs

```diff
@@ -45,6 +48,11 @@ public IEnumerable<WeatherForecast> Get()
         // how dependency calls will be captured and treated
         // automatically as child of incoming request.
         var res = HttpClient.GetStringAsync("http://google.com").Result;
+
+        // Manually create an activity. This will become a child of
+        // the incoming request.
+        using var activity = this.instrumentation.ActivitySource.StartActivity("calculate forecast");
```

Comment:
could we store the activitysource, and counter instead? so that `this.Activitysource.StartActivity`, `this.FreezingDaysCounter` style code can be used.

Reply:
Do you have a preference on the null checks in the constructor?

Reply:
I am just checking for `Instrumentation`

Reply:
oh I see I just made the opposite advice ;p @cijothomas what do you like about the multiple fields? My suggestion would have been to get rid of the "this." but otherwise leave it as-is.